soc/arm/nxp_imx/rt5xx/soc.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Board configuration macros for the MXRT595 platform
+ *
+ * This header file is used to specify and describe board-level aspects for the
+ * 'MXRT595' platform.
+ */
+
+#ifndef _SOC__H_
+#define _SOC__H_
+
+#ifndef _ASMLANGUAGE
+#include <sys/util.h>
+#include <fsl_common.h>
+
+/* Add include for DTS generated information */
+#include <devicetree.h>
```

Comment:
none of these headers is required here.

Reply:
Build breaks if I delete any of these headers.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/41544 and https://github.com/zephyrproject-rtos/zephyr/issues/41543

Reply:
Done, fixed.